### PR TITLE
Minor Improvement:

### PR DESCRIPTION
**Pull request by arturobernalg**
* Use Empty collections
* Unnecessary boxing
* Unnecessary toString
* Use Math.max

### CODE REVIEW

File: httpcore5/src/main/java/org/apache/hc/core5/http/ContentType.java [hunk 1]
```diff
@@ -105,6 +105,11 @@
     public static final ContentType WILDCARD = create(
             "*/*", (Charset) null);
 
+    /**
+     * An empty immutable {@code NameValuePair} array.
+     */
+    public static final NameValuePair[] EMPTY_NAME_VALUE_PAIR_ARRAY = new NameValuePair[0];
```
Comment (ok2c) [inline httpcore5/src/main/java/org/apache/hc/core5/http/ContentType.java:111]: @arturobernalg There is no need to make it public. Please change to private.

**Review by ok2c — CHANGES_REQUESTED** [review]

File: httpcore5/src/main/java/org/apache/hc/core5/http/ContentType.java (hunk 1 above)
Comment (arturobernalg) [reply httpcore5/src/main/java/org/apache/hc/core5/http/ContentType.java]: Ok. changed

**Review by arturobernalg — COMMENTED** [review]
Changed

File: httpcore5/src/main/java/org/apache/hc/core5/http/impl/bootstrap/ServerBootstrap.java [hunk 2]
```diff
@@ -407,7 +407,7 @@ public HttpServer create() {
         }
 
         return new HttpServer(
-                this.listenerPort > 0 ? this.listenerPort : 0,
+                Math.max(this.listenerPort, 0),
```
Comment (ok2c) [inline httpcore5/src/main/java/org/apache/hc/core5/http/impl/bootstrap/ServerBootstrap.java:410]: @arturobernalg What are benefits of using `Math#max` here? Is it more byte code efficient? I am kind of old and stupid. It makes it more difficult for me to understand the intent of the operation. I am also a bit worried that the proposed code appears to make no distinction between greater and greater equal (as well lesser and lesser equal) comparison operations.

Comment (arturobernalg) [reply httpcore5/src/main/java/org/apache/hc/core5/http/impl/bootstrap/ServerBootstrap.java:410]: > @arturobernalg What are benefits of using `Math#max` here? Is it more byte code efficient? I am kind of old and stupid. It makes it more difficult for me to understand the intent of the operation. I am also a bit worried that the proposed code appears to make no distinction between greater and greater equal (as well lesser and lesser equal) comparison operations.

HI @ok2c 
My idea was get more readability. As far as I know for implementation/performance sake, they would be nearly equivalent. Maybe a bit more faster the java implementations. But your right,  no make no distinction between greater and greater equal. I gonna remove that changes.

**Review by ok2c — APPROVED** [review]

**Pull request merged**